runtime/gateway.go

```diff
@@ -739,6 +778,44 @@ func (gateway *Gateway) setupTChannel(config *StaticConfig) error {
 	return nil
 }
 
+func (gateway *Gateway) setupYARPCClientDispatcher(config *StaticConfig) error {
+	ip := config.MustGetString("sidecarRouter.default.grpc.ip")
+	port := config.MustGetInt("sidecarRouter.default.grpc.port")
+	address := fmt.Sprintf("%s:%d", ip, port)
+
+	clientServiceNameMapping := make(map[string]string)
+	config.MustGetStruct("grpc.clientServiceNameMapping", &clientServiceNameMapping)
+
+	// TODO: no op if there is no YARPC clients
+	outbounds := make(yarpc.Outbounds, len(clientServiceNameMapping))
+	for key, value := range clientServiceNameMapping {
+		outbounds[key] = transport.Outbounds{
+			ServiceName: value,
+			// TODO: this setup opens one TCP connection to the sidecar router per client
+			//       Is it better to share the same TCP connection among the clients?
```

Comment:
yup a single transport is intended to support several outbounds, this is what yarpcfx does :)